django/middleware/csrf.py

```diff
@@ -196,6 +227,24 @@ def _set_token(self, request, response):
             # Set the Vary header since content varies with the CSRF cookie.
             patch_vary_headers(response, ('Cookie',))
 
+    def _origin_verified(self, request):
+        good_origin = '%s://%s' % (
+            'https' if request.is_secure() else 'http',
+            request.get_host(),
+        )
+        allowed_origins = [good_origin] + self.allowed_origins_exact
```

Comment:
You could avoid an extra list creation by doing:

```python
allowed_origins = [good_origin, *self.allowed_origins_exact]
```

Or you could avoid any list creations and skip extra request processing in some cases (fast path) by doing:

```python
request_origin = request.META['HTTP_ORIGIN']
if request_origin in self.allowed_origins_exact:
    return True
good_origin = '%s://%s' % (
    'https' if request.is_secure() else 'http',
    request.get_host(),
)
if request_origin == good_origin:
    return True
```

Also, if you do that, you could make `self.allowed_origins_exact` return a set rather than a list since it's being used for a membership test.